presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/ResourceGroupsDao.java

```diff
@@ -93,8 +94,10 @@
     @SqlQuery("SELECT resource_group_id\n" +
             "FROM exact_match_source_selectors\n" +
             "WHERE environment = :environment\n" +
-            "  AND source = :source")
+            "  AND source = :source\n" +
+            "  AND (query_type IS NULL OR query_type = :query_type)")
```

Comment:
The name of this method is slightly confusing.

The fact that this method has name `getExactMatchResourceGroup` sounds like `query_type` must also be an exact match, but the predicate `query_type IS NULL` won't make it an exact match. e.g.,  result of this method can be a resource group with a null query type even though the caller specifies a non-null one.